src/packet/BinaryWriter.js

```diff
@@ -1,137 +1,62 @@
-﻿'use strict';
+'use strict';
 /*
  * Simple BinaryWriter is a minimal tool to write binary stream with unpredictable size.
  * Useful for binary serialization.
  *
- * Copyright (c) 2016 Barbosik
- * 
- * Licensed under the Apache License, Version 2.0 (the "License");
- * you may not use this file except in compliance with the License.
- * You may obtain a copy of the License at
- * 
- *     http://www.apache.org/licenses/LICENSE-2.0
- * 
- * Unless required by applicable law or agreed to in writing, software
- * distributed under the License is distributed on an "AS IS" BASIS,
- * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
- * See the License for the specific language governing permissions and
- * limitations under the License.
- * 
- * 
+ * Copyright (c) 2016 Barbosik https://github.com/Barbosik
+ * License: Apache License, Version 2.0
  */
 
-function BinaryWriter(size) {
-    if (!size || size <= 0) {
-        size = Buffer.poolSize / 2;
-    }
-    this._buffer = new Buffer(size);
+function BinaryWriter() {
+    this._buffer = parseInt(process.version[1]) < 6 ? new Buffer(1e5) : Buffer.allocUnsafe(1e5);

```

Comment:
> Was complete newbie but became advanced at js in half a year.
- I was too. But I learned too.

> Started with OgarUL.
- I started with Ogar.

> Now creates numerous modules and tools
- Recently made spatial hashes. Have a ton of C# and VB projects that can help me out in the future.

> Learning new stuff, such as cryptology.
- "Cryptology"? Is that even a thing? Oh, that's cryptography.

> Learned php and other things in a month.
- About to learn PHP and MySQL.

> Learning other languages for complex projects
- What projects?

> Invented several algorithms.
- What algorithms? Are those private too?

> Learning web design methods.
- "Learning" while arguing with people over small stuff

> Now starting a complex web site called codersAssemble
- "Now"
> Other things such as calculator programs and 3d modeling.
- I know a friend that can make me this.